pkg/async/notifications/email.go

```diff
@@ -18,14 +17,30 @@ const domain = "domain"
 const name = "name"
 const phase = "phase"
 const errorPlaceholder = "error"
+const workflowProject = "workflow.project"
+const workflowDomain = "workflow.domain"
+const workflowName = "workflow.name"
+const workflowVersion = "workflow.version"
+const launchPlanProject = "launch_plan.project"
+const launchPlanDomain = "launch_plan.domain"
+const launchPlanName = "launch_plan.name"
+const launchPlanVersion = "launch_plan.version"
 const replaceAllInstances = -1
 
-func substituteEmailParameters(message string, request admin.WorkflowExecutionEventRequest, execution models.Execution) string {
-	response := strings.Replace(message, fmt.Sprintf(substitutionParam, project), execution.Project, replaceAllInstances)
-	response = strings.Replace(response, fmt.Sprintf(substitutionParam, domain), execution.Domain, replaceAllInstances)
-	response = strings.Replace(response, fmt.Sprintf(substitutionParam, name), execution.Name, replaceAllInstances)
+func substituteEmailParameters(message string, request admin.WorkflowExecutionEventRequest, execution *admin.Execution) string {
+	response := strings.Replace(message, fmt.Sprintf(substitutionParam, project), execution.Id.Project, replaceAllInstances)
```

Comment:
nit, this is kind of verbose and I'm worried error prone because we have to keep these Replace calls in sync with the constants above. can we put all the placeholder string constants in an array defined at the top of the file and then iterate through the array and perform the string substitution in a loop?